packages/sdk/src/chains/Substrate/utils.ts

```diff
@@ -0,0 +1,106 @@
+import { DefinitionRpcExt } from '@polkadot/types/types';
+import { ApiPromise, WsProvider } from '@polkadot/api';
+import { web3Accounts, web3Enable } from '@polkadot/extension-dapp';
+import { keyring as Keyring } from '@polkadot/ui-keyring';
+import { isTestChain } from '@polkadot/util';
+import { TypeRegistry } from '@polkadot/types/create';
+import { ChainType } from '@polkadot/types/interfaces';
+
+import { SubstrateConfigType } from '../../types';
+
+const registry = new TypeRegistry();
+
+///
+// Connecting to the Substrate node
+export const substrateSocketConnect = (
+  state: {
+    apiState: string;
+    socket: string;
+    jsonrpc: {
+      [x: string]: Record<string, DefinitionRpcExt>;
+    };
+  },
+  dispatch: React.Dispatch<{
+    type: string;
+    payload: any;
+  }>,
+): void => {
+  const { apiState, socket, jsonrpc } = state;
+  // We only want this function to be performed once
+  if (apiState) return;
+
+  dispatch({ type: 'CONNECT_INIT', payload: undefined });
+
+  console.log(`Connected socket: ${socket}`);
+  const provider = new WsProvider(socket);
+  const _api = new ApiPromise({ provider, rpc: jsonrpc });
+
+  // Set listeners for disconnection and reconnection event.
+  _api.on('connected', () => {
+    dispatch({ type: 'CONNECT', payload: _api });
+    // `ready` event is not emitted upon reconnection and is checked explicitly here.
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    void _api.isReady.then(_api => dispatch({ type: 'CONNECT_SUCCESS', payload: undefined }));
+  });
+  _api.on('ready', () => dispatch({ type: 'CONNECT_SUCCESS', payload: undefined }));
+  // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
+  _api.on('error', err => dispatch({ type: 'CONNECT_ERROR', payload: err }));
+};
+
+export const retrieveChainInfo = async (
+  api: ApiPromise,
+): Promise<{
+  systemChain: string;
+  systemChainType: ChainType;
+}> => {
+  const [systemChain, systemChainType] = await Promise.all([
+    api.rpc.system.chain(),
+    api.rpc.system.chainType
+      ? api.rpc.system.chainType()
+      : Promise.resolve(registry.createType('ChainType', 'Live') as unknown as ChainType),
```

Comment:
You can declare a type in the outer scope of the function and reuse it, rather than create it on every call or create a registry on the fly.